app/controllers/devise/registrations_controller.rb

```diff
@@ -66,11 +66,23 @@ def update
 
   # DELETE /resource
   def destroy
-    resource.destroy
-    Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
-    set_flash_message! :notice, :destroyed
-    yield resource if block_given?
-    respond_with_navigational(resource){ redirect_to after_sign_out_path_for(resource_name) }
+    if resource.class.require_password_to_destroy
+      resource_destroyed = destroy_resource(resource, account_destroy_params)
+      if resource_destroyed
+        Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
+        set_flash_message! :notice, :destroyed
+        yield resource if block_given?
+        respond_with_navigational(resource){ redirect_to after_sign_out_path_for(resource_name) }
+      else
+        render :edit
+      end
+    else
+      resource.destroy
+      Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
```

Comment:
WDYT about creating a method to extract this sign out logic? That way we can use it in both scenarios, without the need to duplicate it.